src/components/Box/index.tsx

```diff
@@ -1,52 +1,50 @@
-import { Theme, css } from '@emotion/react'
-import styled from '@emotion/styled'
+import { css } from '@emotion/react'
+import styled, { StyledComponent } from '@emotion/styled'
 import { x } from '@xstyled/emotion'
 import PropTypes from 'prop-types'
 import React, {
+  ClassAttributes,
   ElementType,
-  FunctionComponent,
-  ReactNode,
+  HTMLAttributes,
   forwardRef,
 } from 'react'
 
-const borderedStyles = ({ theme }: { theme: Theme }) => css`
-  padding: ${theme.space['3']};
-  border-radius: ${theme.radii.default};
-  border: 1px solid ${theme.colors.gray350};
-`
-
-const StyledBox = styled(x.div, {
-  shouldForwardProp: prop => !['bordered'].includes(prop.toString()),
-})<{ bordered?: boolean }>`
-  ${({ bordered }) => (bordered ? borderedStyles : null)}
-`
-
-type Props = {
+type BoxProps = {
   bordered?: boolean
   width?: number | string
   height?: number | string
-  children?: ReactNode
-} & XStyledProps
+}
 
-const forwardType = forwardRef<Element, Props>(() => null)
+const StyledBox = styled(x.div, {
+  shouldForwardProp: prop => prop !== 'bordered',
+})<BoxProps>`
+  ${({ bordered, theme }) =>
+    bordered
+      ? css`
+          padding: ${theme.space['3']};
+          border-radius: ${theme.radii.default};
+          border: 1px solid ${theme.colors.gray350};
+        `
+      : null}
+`
 
-type BoxType = typeof forwardType & {
-  withComponent?: (
-    element: string | ElementType<unknown>,
-  ) => FunctionComponent<Record<string, unknown>>
-}
+type AllBoxProps = BoxProps &
+  XStyledProps &
+  HTMLAttributes<Element> &
+  ClassAttributes<Element>
 
-const Box: BoxType = forwardRef<Element, Props>(
-  ({ width, height, bordered = false, ...props }, ref) => (
-    // @ts-expect-error As we won't know the Element kind we can't assume that Ref will be a Element
-    <StyledBox ref={ref} w={width} h={height} bordered={bordered} {...props} />
+const Box = forwardRef(
+  // map props `width` to w & `height` to h, see https://xstyled.dev/docs/upgrade-guide/#width-becomes-w-height-becomes-h
+  ({ width, height, ...props }: BoxProps, ref) => (
+    // @ts-expect-error As we won't know the Element kind we can't assume what Ref will be
+    <StyledBox ref={ref} w={width} h={height} {...props} /> // eslint-disable-line react/jsx-props-no-spreading
```

Comment:
We tend to not disable known eslint warnings as we want them to be explicit. This allows us to track them and keep in mind that we have to correct them :)

Reply:
done, ammended c1ba7c0